prow/kube/types.go

```diff
@@ -153,6 +154,16 @@ type ConfigMapKeySelector struct {
 	Key  string `json:"key,omitempty"`
 }
 
+type Resources struct {
+	Requests *ResourceRequest `json:"requests,omitempty"`
+	Limits   *ResourceRequest `json:"limits,omitempty"`
+}
+
+type ResourceRequest struct {
```

Comment:
Code simplicity, git repo size, and binary size, in descending order. Go is excellent for writing simple REST clients, and rolling our own lets us fine-tune without writing a wrapper package.